tools/ci.sh

```diff
@@ -100,6 +100,8 @@ function ci_esp32_build {
     make ${MAKEOPTS} -C mpy-cross
     make ${MAKEOPTS} -C ports/esp32 submodules
     make ${MAKEOPTS} -C ports/esp32
+    make ${MAKEOPTS} -C ports/esp32 clean
+    make ${MAKEOPTS} -C ports/esp32 USER_C_MODULES=/home/runner/work/micropython/micropython/examples/usercmodule/micropython.cmake
```

Comment:
I tried to user ../../examples/usercmodule/micropython.cmake and examples/usercmodule/micropython.cmake here and both failed.  

I'm not sure why the behavior is different here for esp32 versus the other ports that are able to refer to the file in a relative manner.

Reply:
It could be because of the location of the build set by `idf.py`.

The idea with this script/functions is that it can be executed locally, which won't work with this change.  Could you try `$PWD/../../examples/usercmodule/micropython.cmake` and also `$(pwd)/../../examples/usercmodule/micropython.cmake`?

Reply:
See #7091 for a "fix" to this to use relative paths.  The issue was that the main CMakeLists.txt is in `ports/esp32/main/` so that's the directory from where relative paths are taken.